File: agriconnectfrontend/src/services/cartService.js
import axios from "axios";

const API_URL = "http://localhost:3130/api/cart";

// Get the user's cart
export const getCart = (userId) => axios.get(`${API_URL}/${userId}`);

// Add an item to the cart (query params for Spring Boot)
export const addItemToCart = (userId, productId, quantity) =>
  axios.post(`${API_URL}/${userId}/add`, null, { params: { productId, quantity } });

// Update item quantity
export const updateItemQuantity = (userId, productId, quantity) =>
  axios.put(`${API_URL}/${userId}/update`, null, { params: { productId, quantity } });

// Remove item from cart
export const removeItemFromCart = (userId, productId) =>
  axios.delete(`${API_URL}/${userId}/items/${productId}`);

export const clearCart = (userId) =>
  axios.delete(`${API_URL}/${userId}/clear`);

File: agriconnectbackend/src/main/java/com/project/service/CartService.java
package com.project.service;

import com.project.dto.CartDto;

public interface CartService {
    CartDto getCartByUserId(Long userId);
    CartDto addItemToCart(Long userId, Long productId, int quantity);
    CartDto updateItemQuantity(Long userId, Long productId, int quantity);
    void clearCart(Long userId);
}
